web-extensions/chrome/engine/bookmycomics.js

```diff
@@ -37,66 +56,105 @@ BmcEngine.prototype.events = {
     },
 };
 
+/**
+ * Utility function to dispatch the comicId load completion event.
+ */
+BmcEngine.prototype._dispatchLoad = function () {
+    console.warn(`Dispatching: ${this.events.load}`);
+    this.dispatchEvent(new CustomEvent(this.events.load));
+}
+
+/**
+ * Utility function that loads the comic's ID if any, and memorizes/caches it
+ * (hence memoize) to accelerate later requests.
+ *
+ * If a memoization (active loading) is ongoing, no new request will be made,
+ * but the completion event will still notify the beforehand registered
+ * eventListener that the comicId is resolved.
+ * This method allows to offer an unique behavior based on event communication,
+ * whether the comicId was already resolved or not.
+ *
+ */
+BmcEngine.prototype._memoizeComic = function () {
+    if (!this._comic.name) {
```

Comment:
Btw, I don't like much this way of checking things in JS. If name is equal to `"null"` or `"0"`, it'll enter in this condition as well, even if this is the comic's name. If I understand your code correctly, the correct check should be:

```
if (this._comic.name === undefined || this._comic.name.length === 0) {
```

Reply:
Well,numeric 0 and null value should not  be possible anyways.

I can enforce this by checking against undefined, null, non-string and empty string